File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/digitaltwin/event/EventData.java
package org.opcfoundation.webserver.digitaltwin.event;

import org.eclipse.milo.opcua.stack.core.types.builtin.DateTime;
import org.eclipse.milo.opcua.stack.core.types.builtin.Variant;

import java.time.LocalDateTime;
import java.time.ZoneOffset;
import java.util.HashMap;
import java.util.Map;

public class EventData {
    private final String eventId;
    private DateTime time;
    private String message;

    private final Map<String, Variant> eventData;

    public EventData(String eventId)
    {
        eventData = new HashMap<>();
        this.eventId = eventId;
        this.time = DateTime.NULL_VALUE;
        this.message = "";
    }

    public String getEventId() {
        return eventId;
    }

    public DateTime getTime() {
        return time;
    }

    public void setTime(LocalDateTime time) {
        this.time = new DateTime(time.atZone(ZoneOffset.UTC).toInstant());
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public void setFieldData(String field, Variant data) {
        eventData.put(field, data);
    }

    public Map<String, Variant> getEventData() {
        return eventData;
    }
}

File: opcua-webservice-node/source/src/server/digitaltwin/event/EventData.ts
import { UaVariant } from "opcua-webapi-ts";

export class EventData {
    private readonly _eventId: string;
    private _time: Date | null;
    private _message: string;
    private readonly _eventData: Map<string, UaVariant>;

    constructor(eventId: string) {
        this._eventId = eventId;
        this._time = null;
        this._message = "";
        this._eventData = new Map<string, UaVariant>();
    }

    get eventId(): string {
        return this._eventId;
    }

    get time(): Date | null {
        return this._time;
    }

    set time(time: Date) {
        this._time = time;
    }

    get message(): string {
        return this._message;
    }

    set message(message: string) {
        this._message = message;
    }

    setFieldData(field: string, data: UaVariant): void {
        this._eventData.set(field, data);
    }

    get eventData(): Map<string, UaVariant> {
        return this._eventData;
    }
}
